client/src/main/java/io/pravega/client/stream/impl/ClientFactoryImpl.java

```diff
@@ -239,4 +243,31 @@ public void close() {
         connectionFactory.close();
     }
 
+    public void validateStartAndEndStreamCuts(final ReaderGroupConfig readerGroupConfig) {
+        readerGroupConfig.getEndingStreamCuts().entrySet().stream().filter(e -> !e.getValue().equals(StreamCut.UNBOUNDED))
+                .forEach(e -> {
+                    if (readerGroupConfig.getStartingStreamCuts().get(e.getKey()) != StreamCut.UNBOUNDED) {
+                        verifyStartAndEndStreamCuts(readerGroupConfig.getStartingStreamCuts().get(e.getKey()), e.getValue());
+                    }
+                });
+    }
+
+    private void verifyStartAndEndStreamCuts(final StreamCut startStreamCut, final StreamCut endStreamCut) {
+        final Map<Segment, Long> startPositions = startStreamCut.asImpl().getPositions();
+        final Map<Segment, Long> endPositions = endStreamCut.asImpl().getPositions();
+
+        //* verify both stream cut satisfy the "cover full key space" and "no key space overlap" in each of the streamcuts
+        Preconditions.checkArgument(controller.isStreamCutValid(startStreamCut).equals(true),
+                "startStreamCut key space violation.");
+        Preconditions.checkArgument(controller.isStreamCutValid(endStreamCut).equals(true),
+                "endStreamCut key space violation.");
+
+        //* check if the offsets are legal
+        Preconditions.checkArgument(startPositions.entrySet().stream().allMatch(s ->
```

Comment:
this may not be sufficient. as suggested earlier, add a new grpc api on controller that takes two streamcuts and responds if start is strictly less than end.